cmd/incusd/api_project.go

```diff
@@ -1541,3 +1566,93 @@ func projectValidateRestrictedSubnets(s *state.State, value string) error {
 
 	return nil
 }
+
+// Do I need to check the StatusCode and/or ExpandedConfig similar to "cmd/incus/delete.go"
+// Do I need to check if this a snapshot or of type container similar to "cmd/incusd/instance_delete.go"
+// Delete all project instances
+func projectDeleteInstances(ctx context.Context, project *cluster.Project, tx *db.ClusterTx) (error) {
+	instances, err := cluster.GetInstances(ctx, tx.Tx(), cluster.InstanceFilter{Project: &project.Name})
+	if err != nil {
+		return err
+	}
+
+	if len(instances) == 0 {
+		return nil
+	}
+
+	for _, instance := range instances {
+		if err := cluster.DeleteInstance(ctx, tx.Tx(), project.Name, instance.Name); err != nil {
+			return err
+		}
+	}
```

Comment:
Note that this is only deleting the database records so that's not going to work.

You instead need to load the instance, make sure it's stopped and then call `inst.Delete(true)` to have it and all its snapshots properly deleted.

Reply:
fixed in code